src/cirrus/plugins/creds/keyring.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+"""
+keyring module plugin
+
+"""
+import keyring
+from cirrus.creds_plugin import CredsPlugin
+
+
+class Keyring(CredsPlugin):
+    """
+    Keyring
+
+    Get credentials from the platform's keyring implementation
+    Eg: Keychain on OSX
+    See https://pypi.python.org/pypi/keyring for more details
+
+    """
+    PLUGGAGE_OBJECT_NAME = 'keyring'
+
+    def __init__(self):
+        super(Keyring, self).__init__()
+        self.section = "cirrus"
+        self.keyring = None
+
+    def load(self):
+        """
+        """
+        self.keyring = keyring.get_keyring()
+
+    def github_credentials(self):
```

Comment:
Yeah, I think if I end up adding another plugin, thats a good idea, but keyring seems to cover a bunch of common ones, so will leave it explicit for now